File: Selenium/Project/Activity6.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.time.Duration;


public class Activity6 {
    WebDriver driver;

    @BeforeClass
    public void beforeMethod() {
        // Set up Firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        // Create the Wait object
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        // Create the Actions object
        Actions builder = new Actions(driver);

        // Open the page
        driver.get("https://alchemy.hguy.co/jobs/");
        // Print the title of the page

    }
    @Test
    public void Activity6() {

            driver.findElement(By.linkText("Jobs")).click();
            String secondPageTitle = driver.getTitle();
            System.out.println("Job Page Title:" + secondPageTitle);
            driver.findElement(By.xpath("//input[@id='search_keywords']")).sendKeys("Banking");
            driver.findElement(By.xpath("//input[@id='search_location']")).sendKeys("Banglore");
            driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(20));
            driver.findElement(By.xpath("//input[@type= 'submit' and @value = 'Search Jobs']")).click();
            String profile = driver.findElement(By.xpath("/html/body/div/div/div/div/main/article/div/div/ul/li/a/div[1]/h3")).getText();
            System.out.println(profile);
            if (profile.equals("SAP Testing")) {
                // Close the Browser
                System.out.println("Page Heading matches");
                driver.findElement(By.xpath("/html/body/div/div/div/div/main/article/div/div/ul/li/a")).click();
                //Assert.assertEquals(By.class"entry-title")).get;
            } else
                System.out.println("Page heading doesn't matches");
            driver.findElement(By.xpath("//input[@type = 'button' and @value = 'Apply for job']")).click();
            //driver.findElement(By.xpath("//button[text()='button']")).click();
            WebElement email = driver.findElement(By.className("job_application_email"));
            driver.findElement(By.cssSelector(".job_application_email")).click();
            System.out.println(email);
        }
    }

File: Selenium/Project/Activity6.py
import pytest
from selenium import webdriver
from selenium.webdriver.common.by import By


# Create a fixture to initialize our webdriver
@pytest.fixture(scope="session")
def setup(request):
    # Initialize the webdriver
    driver = webdriver.Firefox()

    # Get the underlying collection
    session = request.node

    # Pass through the driver to the test class
    for item in session.items:
        cls = item.getparent(pytest.Class)
        setattr(cls.obj, "driver", driver)

    # Execute the test method
    yield

    # Close the browser once the test has ended
    request.addfinalizer(driver.close)


@pytest.mark.usefixtures("setup")
class TestSite:

    def test_applyjob(self):
        self.driver.get("https://alchemy.hguy.co/jobs")

        Jobslink = self.driver.find_element(By.LINK_TEXT, "Jobs")
        Jobslink.click()
        #Jobsection = self.driver.find_element(By.CLASS_NAME, "entry-title")
        #print("Heading: ", Jobsection.text)
        GetTitle = self.driver.title
        Expected_title = "Jobs – Alchemy Jobs"

        assert GetTitle == Expected_title, "Title mismatch"
        search = self.driver.find_element(By.XPATH, "//input[@id='search_keywords']")
        place =  self.driver.find_element(By.XPATH, "//input[@id='search_location']")
        search.send_keys("Banking")
        place.send_keys("Banglore")
        #SearchJobs = self.driver.find_element(By.LINK_TEXT, "Search Jobs")
        SearchJobs = self.driver.find_element(By.XPATH, "//input[@type= 'submit' and @value = 'Search Jobs']")
        print(SearchJobs)
        SearchJobs.click()
        self.driver.implicitly_wait(90)
        profile = self.driver.find_element(By.XPATH, "/html/body/div/div/div/div/main/article/div/div/ul/li/a/div[1]/h3")
        print(profile.text)
        profile.click()
        self.driver.implicitly_wait(90)
        ApplyJob = self.driver.find_element(By.XPATH, "//input[@type= 'button' and @value = 'Apply for job']")
        print(ApplyJob)
        ApplyJob.click()
        self.driver.implicitly_wait(3600)
        Jobe = self.driver.find_element(By.CLASS_NAME, "job_application_email")
        print(Jobe)
        Jobe.click()
        self.driver.implicitly_wait(3600)



